src/html5/node.rs

```diff
@@ -121,13 +122,18 @@ pub struct Node {
     pub namespace: Option<String>,
     /// actual data of the node
     pub data: NodeData,
-    /// pointer to document this node is attached to
-    pub document: DocumentHandle,
-
+    /// weak pointer to document this node is attached to
+    pub document: Weak<RefCell<Document>>,
     // Returns true when the given node is registered into an arena
     pub is_registered: bool,
 }
 
+impl PartialEq for Node {
+    fn eq(&self, other: &Node) -> bool {
+        self.id == other.id
+    }
+}
+
```

Comment:
This was added since `Weak<RefCell>` doesn't have a `PartialEq` implementation and `#[derive(PartialEq)]` no longer worked. I think two Nodes should be equal if their `id`s are equal